torch/export/export_schema.py

```diff
@@ -0,0 +1,326 @@
+    #   // Data can be stored inside the protobuf file using type-specific fields or raw_data.
+    #   // Alternatively, raw bytes data can be stored in an external file, using the external_data field.
+    #   // external_data stores key-value pairs describing data location. Recognized keys are:
+    #   // - "location" (required) - POSIX filesystem path relative to the directory where the ONNX
+    #   //                           protobuf model was stored
+    #   // - "offset" (optional) - position of byte at which stored data begins. Integer stored as string.
+    #   //                         Offset values SHOULD be multiples 4096 (page size) to enable mmap support.
+    #   // - "length" (optional) - number of bytes containing data. Integer stored as string.
+    #   // - "checksum" (optional) - SHA1 digest of file specified in under 'location' key.
+    #   repeated StringStringEntryProto external_data
+    location: str
+    offset: str     # !!! Consider using int, but int has int_max limitation
+    length: str     # !!! Consider using int, but int has int_max limitation
+    checksum: str
+
+
+@dataclass
+class Storage:
+    class DataLocation(Enum):
+        Default = auto()
+        External = auto()
+
+    data_location: DataLocation
+    data: Union[Buffer, ExternalBuffer]
+
+
+# This is a concrete tensor backed by storage
+@dataclass
+class Tensor:
+    # storage
+    storage: Storage
+
+    # metadata
+    meta: TensorMeta
+
+
+# !!! consider adding an optional named field to Tensor directly to avoid defining a new class
+@dataclass
+class NamedTesnsor:
+    name: str
+    tensor: Tensor
+
+################################################################################
+# Following section is the defining the schema of 3 level construct: GraphModule, Graph, Node
+
+# IValue has no corresponding class in fx
+# IValue is the "values" that are passed between nodes in the graph
+# !!! Assumption: Only Tensor can be passed between nodes, and not other int/float/bool types...
+# IValue is a named virtual tensor, with an optional TensorMeta that describes the properties of the tensor
+# !!! Consider using a more descriptive name, e.g. TensorValue, TensorPlaceholder, TensorArgument, etc.
+@dataclass
+class IValue:
+    name: str   # A unique identifider name for the IValue
+                # The name will be used in the graph and node to refer to the IValue
+
+    meta: TensorMeta
+
+
+# Maps to fx.Node
+@dataclass
+class Node:
+    op: str         # One of ['placeholder', 'call_function', 'get_attr', 'output'],
+                    # call_method and call_module are not supported, as they shouldn't apprear in the Caononical FX Graph
+                    # !!! Consider using an enum instead of string
+
+    target: str      # fully qualified name to the target, e.g. aten.add.Tensnor
+                     # !!! Consider using a structured operator name instead of string
+
+    args: List[Argument]              # args for this node
```

Comment:
I think args should be List[Union[Node, Value]] like it is now.

Reply:
why? seems simpler to just args consume output values always